### DISCUSSION TITLE
error: database "admin" does not exist

### DISCUSSION THREAD

**Discussion by 5t33** [tada5hi/typeorm-extension#401] [Q&A]
I'm a bit confused. I'm attempting to create a database using 
```
./node_modules/typeorm-extension/dist/cli/index.js db:create -d ormconfig.ts --initialDatabase <name of my database>
```
with ormConfig like so:
```
import { DataSource } from 'typeorm';

export const connectionSource = new DataSource({
    migrationsTableName: 'migrations',
    type: 'postgres',
    host: 'localhost',
    port: 5432,
    username: 'admin',
    password: 'admin',
    database: '<my database>',
    logging: false,
    synchronize: false,
    name: 'default',
    entities: ['src/**/**.entity{.ts,.js}'],
    migrations: ['src/migrations/**/*{.ts,.js}'],
    subscribers: ['src/subscriber/**/*{.ts,.js}'],
    dropSchema: true,
});
```

but getting `error: database "<my database>" does not exist`

There isn't a database because I'm trying to create one. Am I missing some basic postgres knowledge? Like there needing to be ad admin db or something?

**Comment by tada5hi** [comment]
@5t33 your assumption is right. Postgres behaves a bit special here. It requires an `initialDatabase` to connect to. By default the value is equal to the user name. 
For more details: https://github.com/tada5hi/typeorm-extension#options
